airbyte-ci/connectors/pipelines/pipelines/helpers/changelog.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+import re
+from operator import attrgetter
+from pathlib import Path
+
+import semver
+from typing_extensions import Set
+
+
+class Changelog:
+    def __init__(self, changelog_file_path: Path) -> None:
+        self.changelog_file_path = changelog_file_path
+        if not changelog_file_path.exists():
+            raise Exception(f"File {changelog_file_path} does not exist")
+        if not changelog_file_path.is_file():
+            raise Exception(f"File {changelog_file_path} is not a regular file")
+        self.original_markdown_lines = changelog_file_path.read_text().splitlines()
+        self.__parse_markdown()
+        self.new_entries: Set[ChangelogEntry] = set()
+
+    def add_entry(self, version: semver.Version, date: datetime.date, pull_request_number: int, comment: str) -> None:
+        self.new_entries.add(ChangelogEntry(date, version, pull_request_number, comment))
```

Comment:
In Python sets are not ordered, so you can have surprises, but I think it's not a problem as sorting happens in `to_markdown`.

Reply:
yes, that was the whole idea.